### ISSUE TITLE
More options for `max-len` rule

### ISSUE THREAD

**Issue by tunnckoCore** [eslint/eslint#8114]
**What rule do you want to change?**
`max-len`

**Does this change cause the rule to produce more or fewer warnings?**
neither more or fewer, it would be disabled by default.

**How will the change be implemented? (New option, new default behavior, etc.)?**
Two new options called `ignoreOwnLineComments` and `ignoreBannerComments`. We have `ignoreComments` and `ignoreTrailingComments`, but we don't have for the last case - ignoring thse comments that are in their own line. It will allow us to just ignore them and not ignore the trailing comments.

More better would be to have at least `ignoreBannerComments` to totally ignore the starting comments, because sometime it can have long license/copyright line, and actually is the intention of opening that issue. I'm just want to stick to max-len: 80 and don't ignore comments at all, but to ignore the first comment that i have in almost all of my ~300 packages and all of my files of these packages.

One solution can be to make `comments` option a bit longer than that defined in `code` option, but then trailing comments will respect that too, but i don't want that.

edited.

**Comment by not-an-aardvark** [comment]
Thanks for the feature request. May you please fill out the [rule change proposal template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md)?

Also, I have a few questions:

* Does `ignoreComments: true`, `ignoreTrailingComments: false` have the same effect as the proposed `ignoreOwnLineComments` option?
* How is a "banner comment" defined?

**Comment by eslintbot** [comment]
Hi @tunnckoCore, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by tunnckoCore** [comment]
I'll clarify that today, okey. :)

**Comment by tunnckoCore** [comment]
> Does ignoreComments: true, ignoreTrailingComments: false have the same effect as the proposed ignoreOwnLineComments option?


I don't believe this combination would work in anyway. But no, don't think it will be the same. Idea is that i don't want to ignore any comments. Just these that are on their own line, and more specifically these that are at the top of the file (where usually we have license, copyright and etc stuff)

> How is a "banner comment" defined?

The very first comment (block comment), at the very beginning of the file.

```js
/*!
 * eslint-config-charlint <https://github.com/tunnckoCore/eslint-config-charlint>
 *
 * Copyright (c) 2017 Charlike Mike Reagent <@tunnckoCore> (https://i.am.charlike.online)
 * Released under the MIT license.
 */

'use strict'

// code below
```

Some times some of the lines in these comments are a bit longer than 80, so i want to ignore these comments, but not the other ones. I just want to strictly have 80 length.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
